Library/Homebrew/install.rb

```diff
@@ -363,6 +380,89 @@ def install_formula(formula_installer)
 
         Upgrade.install_formula(formula_installer, upgrade:)
       end
+
+      def ask_input
+        ohai "Do you want to proceed with the installation? [Y/y/yes/N/n]"
+        accepted_inputs = %w[y yes]
+        declined_inputs = %w[n no]
+        loop do
+          result = $stdin.gets
+          return unless result
+
+          result = result.chomp.strip.downcase
+          if accepted_inputs.include?(result)
+            break
+          elsif declined_inputs.include?(result)
+            exit 1
+          else
+            puts "Invalid input. Please enter 'Y', 'y', or 'yes' to proceed, or 'N' to abort."
+          end
+        end
+      end
+
+      # Build a unique list of formulae to size by including:
+      # 1. The original formulae to install.
+      # 2. Their outdated dependents (subject to pruning criteria).
+      # 3. Optionally, any installed formula that depends on one of these and is outdated.
+      def compute_sized_formulae(formulae, args:)
+        sized_formulae = formulae.flat_map do |formula|
+          # Always include the formula itself.
+          formula_list = [formula]
+
+          deps = args.build_from_source? ? formula.deps.build : formula.deps.required
+          # If there are dependencies, try to gather outdated, bottled ones.
+          if deps.any?
+            outdated_dependents = deps.map(&:to_formula).reject(&:pinned?).select do |dep|
+              dep.installed_kegs.empty? || (dep.bottled? && dep.outdated?)
+            end
+            deps.map(&:to_formula).each do |f|
+              outdated_dependents.concat(f.recursive_dependencies.map(&:to_formula).reject(&:pinned?).select do |dep|
+                dep.installed_kegs.empty? || (dep.bottled? && dep.outdated?)
+              end)
+            end
+            formula_list.concat(outdated_dependents)
+          end
+
+          formula_list
+        end
+
+        # Add any installed formula that depends on one of the sized formulae and is outdated.
+        unless Homebrew::EnvConfig.no_installed_dependents_check?
+          sized_formulae.concat(Formula.installed.select do |installed_formula|
+            installed_formula.bottled? && installed_formula.outdated? &&
+              installed_formula.deps.required.any? { |dep| sized_formulae.include?(dep.to_formula) }
```

Comment:
```suggestion
              installed_formula.deps.required.map(&:to_formula).any? { |dep| sized_formulae.include?(dep) }
```
any may be able to simplify this further still with `intersect?`

Reply:
I forgot this builtin function, thanks !